aikit/automl/worker.py

```diff
@@ -0,0 +1,133 @@
+        infos["seed"] = self.seed or int(md5_hash(infos), 16) % (2 ** 32 - 1)
+
+        self.infos = infos
+        self.storage.save_json(infos, self.uuid, 'workers')
+        self.random_state = check_random_state(infos["seed"])
+        self.scorers = OrderedDict()
+        for i, s in enumerate(self.job_config.scoring):
+            if isinstance(s, str):
+                self.scorers[s] = SCORERS[s]
+            else:
+                self.scorers["scorer_%d" % i] = SCORERS[s]
+
+    def run(self):
+        X, y, *_ = self.data.load_pickle(self.data_key)
+        self.cv = create_cv(self.job_config.cv, y, classifier=self.job_config.is_classification(),
+                            shuffle=True, random_state=123)
+
+        while True:
+            job_id = self.queue.dequeue()
+            if job_id is not None:
+                self.compute_job(job_id, X, y)
+            else:
+                logging.info('Waiting for new job...')
+                time.sleep(5)
+
+    def compute_job(self, job_id, X, y, groups=None):
+        logger.info('Running job {}'.format(job_id))
+        status = {
+            'job_id': job_id,
+            'worker_id': self.uuid,
+            'start_time': time.strftime("%Y-%m-%d %H:%M:%S")
+        }
+        self.storage.save_json(status, job_id, 'running')
+
+        try:
+            job = self.storage.load_special_json(job_id, 'jobs')
+            self.validate_job(job)
+            cv_result, yhat = self.compute_cv(job, X, y, groups)
+            status['results'] = cv_result.to_dict(orient='records')
+
+            if self.job_config.additional_scoring_function is not None:
+                additional_result = self.job_config.additional_scoring_function(cv_result, yhat, y, groups)
+                status['additional_results'] = additional_result.to_dict(orient='records')
+
+            if not self.job_config.is_clustering():
+                status['train_metric'] = cv_result["train_%s" % self.job_config.main_scorer].mean()
+                status['test_metric'] = cv_result["test_%s" % self.job_config.main_scorer].mean()
+                logger.info('Train {}: {:.2%}'.format(self.job_config.main_scorer, status['train_metric']))
+                logger.info('Test {}: {:.2%}'.format(self.job_config.main_scorer, status['test_metric']))
+
+            if self.job_config.is_clustering():
+                status['labels'] = yhat
+
+        except Exception as e:
+            if isinstance(e, MemoryError):
+                gc.collect()
+
+            status['error'] = repr(e)
+            status['traceback'] = traceback.format_exc()
+            logger.error(status['error'])
+
+        finally:
+            self.storage.remove(job_id + '.json', 'running')
+            status['end_time'] = time.strftime("%Y-%m-%d %H:%M:%S")
+            self.storage.save_json(status, job_id, 'finished')
+
+    def validate_job(self, job):
+        if job['cv_type'] not in ["approximate", "full"]:
```

Comment:
again while we refactoring maybe read name from a small class
class TypeOfJob:
    approximate="approximate"
    full="full"

like what is done in "enum.py"

that way 
1. Avoid typing mistake
2. make clear it is a fix list of value (not just any string)
3. but still stored as a string (we can't use a real enum class because we'll want to json it)

Reply:
Done